lib/src/rules/prefer_final_fields.dart

```diff
@@ -64,6 +64,27 @@ class GoodMutable {
 ```
 ''';
 
+bool _isMutated(
+        VariableDeclaration variable, CompilationUnit compilationUnit) =>
+    DartTypeUtilities
+        .traverseNodesInDFS(compilationUnit)
+        .where((n) =>
+            n is SimpleIdentifier && n.bestElement is PropertyAccessorElement)
+        .any((n) {
+      SimpleIdentifier identifier = n as SimpleIdentifier;
+      PropertyAccessorElement bestElement =
+          identifier.bestElement as PropertyAccessorElement;
+      if (bestElement.variable != variable.element) {
+        return false;
+      }
+
+      return identifier.getAncestor((a) =>
+              a is AssignmentExpression ||
+              a is PrefixExpression ||
+              a is PostfixExpression) !=
+          null;
```

Comment:
It was not an answer to that comment, it was a previous comment to yours. [This](https://github.com/dart-lang/linter/pull/508#discussion_r107043160) was my answer to your comment in line 82.